src/Main.elm

```diff
@@ -1,10 +1,198 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser exposing (sandbox)
+import Data.Audience exposing (..)
+import Data.AudienceFolder exposing (..)
+import Dict exposing (Dict)
+import Dict.Extra as DE exposing (fromListDedupe)
+import Html exposing (Html, div, button, text)
+import Html.Events
+import Html.Attributes
+import Json.Decode as JD exposing (decodeString)
+import List as L exposing (map, drop, filter)
+import Maybe as M
 
 
-main : Html msg
-main =
-    Html.text "There will be app soon!"
+type alias Model = {
+    audiences : Dict Int (List Audience)
+    , audienceFolders :  Dict Int (List AudienceFolder)
+    , currentFolder : List Int
```

Comment:
It would be nice to have somehow documented the order of things that live in this `currentFolder` field (is it newest first, or oldest first?)

Perhaps even using a different data structure would make it more obvious (`: Stack Int` - stacks are LIFO) but it might be an overkill so I'd be fine with `-- newest first`-like comment.

BTW the name of the field could be more descriptive - something like `breadcrumbs`? `pathToRoot`? `currentFolder` suggests there's just one item but in fact it's a list...

Reply:
You are right, it should be obvious whether it is newest first or oldest first. I used the newest first because prepending in Haskell is cheaper than appending. Therefore I thought, that in Elm too. And the name of the field could be more descriptive, I agree.

Reply:
Yeah! To be fair, I think it would be safe to bet that the list will grow/shrink at the beginning just because of how linked lists work.

Still, I think that when getting into the code after 2 months it's nice to be able to get up to speed with a well placed comment / type name / ... than having to Debug.log it to be sure or to look at how it's updated in `update`. :slightly_smiling_face: